spec/requests/admin/badges_controller_spec.rb

```diff
@@ -244,6 +260,114 @@
 
         expect(response.status).to eq(422)
       end
+
+      context "when grant_existing_holders is true" do
+        it "fails when the badge cannot be granted multiple times" do
+          file = file_from_fixtures('user_emails.csv', 'csv')
+          badge.update!(multiple_grant: false)
+          post "/admin/badges/award/#{badge.id}.json", params: {
+            file: fixture_file_upload(file),
+            grant_existing_holders: true
+          }
+
+          expect(response.status).to eq(422)
+          expect(response.parsed_body['errors']).to eq([
+            I18n.t("badges.mass_award.errors.cant_grant_multiple_times", badge_name: badge.name)
+          ])
+        end
+
+        it "fails when CSV file contains more entries that it's allowed" do
+          badge.update!(multiple_grant: true)
+          csv = Tempfile.new
+          csv.write("#{user.username}\n" * 101)
```

Comment:
If we're already stubbing the constant, let's stub it to a small number to avoid allocating a larger than required string.